lib/Vanguard.cpp

```diff
@@ -1,51 +1,203 @@
+    llvm::initializeHardwareLoopsPass(registry);
+    llvm::initializeTypePromotionPass(registry);
+    llvm::initializeReplaceWithVeclibLegacyPass(registry);
+
+    llvm::cl::ParseCommandLineOptions(argc, argv, "Vanguard Static Analyzer\n");
+
+    llvm::Optional<llvm::PGOOptions> P = llvm::None;
+
+    llvm::LoopAnalysisManager LAM;
+    llvm::FunctionAnalysisManager FAM;
+    llvm::CGSCCAnalysisManager CGAM;
+    llvm::ModuleAnalysisManager MAM;
+
+    llvm::PassInstrumentationCallbacks PIC;
+    llvm::PrintPassOptions PrintPassOpts;
+    PrintPassOpts.Verbose = true;
+    PrintPassOpts.SkipAnalyses = false;
+    llvm::StandardInstrumentations SI(true, true, PrintPassOpts);
+
+    llvm::PipelineTuningOptions PTO;
+
+    PTO.LoopUnrolling = false;
+
+    /*llvm::Triple ModuleTriple(modules[0]->getTargetTriple());
+    std::string CPUStr, FeaturesStr;
+    llvm::TargetMachine *Machine = nullptr;
+    const llvm::TargetOptions Options =
+            llvm::codegen::InitTargetOptionsFromCodeGenFlags(ModuleTriple);
+
+    if (ModuleTriple.getArch()) {
+        CPUStr = llvm::codegen::getCPUStr();
+        FeaturesStr = llvm::codegen::getFeaturesStr();
+        Machine = GetTargetMachine(ModuleTriple, CPUStr, FeaturesStr, Options);
+    } else if (ModuleTriple.getArchName() != "unknown" &&
+               ModuleTriple.getArchName() != "") {
+        llvm::errs() << argv[0] << ": unrecognized architecture '"
+                     << ModuleTriple.getArchName() << "' provided.\n";
+        return 1;
     }
 
-    bool registerVanguardModulePasses(llvm::StringRef name, llvm::ModulePassManager &passMgr, llvm::ArrayRef<llvm::PassBuilder::PipelineElement>) {
-        if (name == "statGen") {
-            passMgr.addPass(vanguard::StatGen());
-            return true;
+    std::unique_ptr<llvm::TargetMachine> TM(Machine);*/
+    llvm::PassBuilder PB(nullptr, PTO, P, &PIC);
+
+    llvm::AAManager AA;
+    if (auto Err = PB.parseAAPipeline(AA, "default")) {
+        return 1;
+    }
+
+    FAM.registerPass([&] { return std::move(AA); });
+
+    // Register all the basic analyses with the managers.
+    PB.registerModuleAnalyses(MAM);
+    PB.registerCGSCCAnalyses(CGAM);
+    PB.registerFunctionAnalyses(FAM);
+    PB.registerLoopAnalyses(LAM);
+    PB.crossRegisterProxies(LAM, FAM, CGAM, MAM);
+
+    llvm::ModulePassManager MPM;
+
+    llvm::cl::PrintOptionValues();
+
+    llvm::LLVMContext ctxt;
+
+    llvm::SMDiagnostic Err;
+
+    ctxt.setDiscardValueNames(false);
+
+    auto setDataLayout = [](llvm::StringRef) -> llvm::Optional<std::string> {
+        return llvm::None;
+    };
+
+    auto &llvmTrans = vanguard::LLVMtoVanguard::getInstance();
+
+    std::vector<std::unique_ptr<llvm::Module>> modules;
+    std::vector<vanguard::CompilationUnit *> units;
+    for (auto inFile: inputFiles) {
+        auto module = parseIRFile(inputFiles[0], Err, ctxt, setDataLayout);
+        if (!module) {
+            Err.print(argv[0], llvm::errs());
+            return 1;
```

Comment:
super minor nitpick:
```suggestion
            return EXIT_FAILURE;
```